packages/models-library/src/models_library/healthchecks.py

```diff
@@ -0,0 +1,28 @@
+#
+# healthcheck models for liveness probes and readiness probes
+#
+# SEE https://medium.com/polarsquad/how-should-i-answer-a-health-check-aa1fcf6e858e
+# SEE https://docs.docker.com/engine/reference/builder/#healthcheck
+
+
+from datetime import timedelta
+from typing import TypeAlias
+
+from pydantic import BaseModel
+
+
+class IsResponsive(BaseModel):
+    elapsed: timedelta  # time elapsed to respond
+
+    def __bool__(self) -> bool:
+        return True
+
+
+class IsNonResponsive(BaseModel):
```

Comment:
question: what is the idea between having these 2 classes vs 1 response with a field giving the responsiveness result?

Reply:
I am separating the representable states and creating a **sum type** . Each representable state has different information but evaluate as true (when successful) and false (when failed). 
There well be more states when we introduce outcomes for readiness (these are for liveness) ... 

SEE some details in "Products and Sum Types" in Robust Python by P. Viafore